ant-protocol/src/error.rs

```diff
@@ -82,3 +82,107 @@ impl From<store::Error> for Error {
         Error::RecordParsingFailed
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_error_retro_compatibility() {
+        // Test with a new struct that has a new variant
+        #[derive(Error, Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+        #[non_exhaustive]
+        enum ExtendedError {
+            #[error("Chunk does not exist {0:?}")]
+            ChunkDoesNotExist(NetworkAddress),
+            #[error("Failed to deserialize hex ScratchpadAddress")]
+            ScratchpadHexDeserializeFailed,
+            #[error("Failed to derive CipherText from encrypted_data")]
+            ScratchpadCipherTextFailed,
+            #[error("Provided cypher text is invalid")]
+            ScratchpadCipherTextInvalid,
+            #[error("There was an error getting the storecost from kademlia store")]
+            GetStoreQuoteFailed,
+            #[error("There was an error generating the payment quote")]
+            QuoteGenerationFailed,
+            #[error("Peer {holder:?} cannot find Record {key:?}")]
+            ReplicatedRecordNotFound {
+                holder: Box<NetworkAddress>,
+                key: Box<NetworkAddress>,
+            },
+            #[error("Could not Serialize/Deserialize RecordHeader to/from Record")]
+            RecordHeaderParsingFailed,
+            #[error("Could not Serialize/Deserialize Record")]
+            RecordParsingFailed,
+            #[error("The record already exists, so do not charge for it: {0:?}")]
+            RecordExists(PrettyPrintRecordKey<'static>),
+            // New variant that doesn't exist in the original Error enum
+            #[error("New error variant for testing")]
+            NewErrorVariant,
+            #[error("Unknown error variant")]
+            #[serde(other)]
+            Unknown,
+        }
+
+        // Test with a struct that has a missing variant (simulating older version)
```

Comment:
the example doesn't need to have so many entries,
it's even bit difficult to figure out which one got `reduced` .
several entries shall be enough, and a comment to explicitly state which entry got `reduced` will be better.

Reply:
There is a comment for that look below